packages/dds/merge-tree/src/client.ts

```diff
@@ -461,21 +424,6 @@ export class Client {
         let segments: ISegment[] | undefined;
         if (op.seg) {
             segments = [this.specToSegment(op.seg)];
-        } else if (op.register) {
```

Comment:
actually, i wonder if we should keep this condition or something like `register in op` and throw if we hit it, otherwise if in the small chance someone is using it, this will lead to corruption

Reply:
Chatted with Tony offline - this is a code path we already hit prior to this change in the case that the receiving client doesn't have the register populated.  The receiving client ignores it (`segments` is `undefined` so return false), resulting in eventual inconsistency.

By ignoring the op we'll return to an eventually consistent state (all clients will equally ignore it).  Since we already have this path today and the document remains usable after the fact this is probably the safer option.